### DISCUSSION TITLE
Project file generation binding

### DISCUSSION THREAD

**Discussion by wojciech-kulik** [wojciech-kulik/xcodebuild.nvim#34] [Tips & Tricks]
If you are using some tool to generate the project file (like xcodegen or Tuist), it's worth setting up some binding to regenerate the project.

If your project config is not in your root directory, you can use a small trick with `pushd` and `popd` to run your commands as if you were in the right location.

```lua
vim.keymap.set(
  "n",
  "<leader>xg",
  "<cmd>!pushd YOUR_PROJECT_FOLDER && xcodegen generate; popd<cr>",
  { desc = "Run xcodegen" }
)
```

You can use the same approach to update your dependencies like CocoaPods or SPM packages.

**Comment by guitaripod** [comment]
@wojciech-kulik I have the plugin working for most things, but the most important one, our work project using Tuist 4.11 doesn't. I'm not sure how to configure it. Also, are you saying I need to open just some Xcode project and then use tuist generate to have Neovim open it correctly? That seems wrong, so I'm simply misunderstanding.

**Reply by wojciech-kulik** [reply]
Hi, yes you need to have the project generated first to be able to configure the plugin. So just run your Tuist CLI, then open the root directory in Neovim and run `:XcodebuildSetup`

**Reply by guitaripod** [reply]
Thanks, I tried it. This is a monolith with multiple targets and schemes, so had to navigate through the setup screen to find the one that gives me scheme selector. Most of them give an empty list. I found the right one, and now it works. Amazing
